core/tee/tee_svc.c

```diff
@@ -56,11 +56,11 @@ void syscall_log(const void *buf __maybe_unused, size_t len __maybe_unused)
 	kbuf = malloc(len);
 	if (kbuf == NULL)
 		return;
-	*kbuf = '\0';
 
-	/* log as Info/Raw traces */
-	if (tee_svc_copy_from_user(kbuf, buf, len) == TEE_SUCCESS)
-		TAMSG_RAW("%.*s", (int)len, kbuf);
+	if (tee_svc_copy_from_user(kbuf, buf, len) == TEE_SUCCESS) {
+		kbuf[len] = '\0';
+		trace_ext_puts(kbuf);
```

Comment:
I guess that was the reason Philip did something like
```
+# if (TRACE_LEVEL == TRACE_ERROR)
  		TAMSG_RAW("%.*s", (int)len, kbuf);
 +# elif (TRACE_LEVEL <= TRACE_FLOW)
 +		IMSG_RAW("%.*s", (int)len, kbuf);
 +# endif
```
It would be nice if we don't need to ifdef there. What about ifdef in `trace_ext_puts` instead?